lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -86,93 +83,74 @@ abstract class AbstractQuery
      * @var ArrayCollection|Parameter[]
      * @psalm-var ArrayCollection<int, Parameter>
      */
-    protected $parameters;
+    protected ArrayCollection $parameters;
 
     /**
      * The user-specified ResultSetMapping to use.
-     *
-     * @var ResultSetMapping|null
-     */
-    protected $_resultSetMapping;
-
-    /**
-     * The entity manager used by this query object.
-     *
-     * @var EntityManagerInterface
      */
-    protected $_em;
+    protected ?ResultSetMapping $_resultSetMapping = null;
 
     /**
      * The map of query hints.
      *
      * @psalm-var array<string, mixed>
      */
-    protected $_hints = [];
+    protected array $_hints = [];
 
     /**
      * The hydration mode.
      *
-     * @var string|int
      * @psalm-var string|AbstractQuery::HYDRATE_*
      */
-    protected $_hydrationMode = self::HYDRATE_OBJECT;
+    protected string|int $_hydrationMode = self::HYDRATE_OBJECT;
 
-    /** @var QueryCacheProfile|null */
-    protected $_queryCacheProfile;
+    protected ?QueryCacheProfile $_queryCacheProfile = null;
 
     /**
      * Whether or not expire the result cache.
-     *
-     * @var bool
      */
-    protected $_expireResultCache = false;
+    protected bool $_expireResultCache = false;
 
-    /** @var QueryCacheProfile|null */
-    protected $_hydrationCacheProfile;
+    protected ?QueryCacheProfile $_hydrationCacheProfile = null;
 
     /**
      * Whether to use second level cache, if available.
-     *
-     * @var bool
      */
-    protected $cacheable = false;
+    protected bool $cacheable = false;
 
-    /** @var bool */
-    protected $hasCache = false;
+    protected bool $hasCache = false;
 
     /**
      * Second level cache region name.
-     *
-     * @var string|null
      */
-    protected $cacheRegion;
+    protected ?string $cacheRegion = null;
 
     /**
      * Second level query cache mode.
      *
-     * @var int|null
      * @psalm-var Cache::MODE_*|null
      */
-    protected $cacheMode;
+    protected ?int $cacheMode = null;
 
-    /** @var CacheLogger|null */
-    protected $cacheLogger;
+    protected ?CacheLogger $cacheLogger = null;
 
-    /** @var int */
-    protected $lifetime = 0;
+    protected int $lifetime = 0;
 
     /**
      * Initializes a new instance of a class derived from <tt>AbstractQuery</tt>.
      */
-    public function __construct(EntityManagerInterface $em)
-    {
-        $this->_em        = $em;
+    public function __construct(
+        /**
+         * The entity manager used by this query object.
+         */
+        protected EntityManagerInterface $_em
```

Comment:
I'd rather rename the property to `em` than adding that `_` prefix to the constructor parameter, tbh.

Reply:
Me too, that was me not questioning Rector's work.